src/box/index.cc

```diff
@@ -559,6 +603,113 @@ iterator_next_internal(struct iterator *it, struct tuple **ret)
 	return it->next_internal(it, ret);
 }
 
+int
+iterator_position(struct iterator *it, const char **pos, uint32_t *size)
+{
+	assert(it->position != NULL);
+	assert(pos != NULL);
+	assert(size != NULL);
+	if (!iterator_is_valid(it)) {
+		*pos = NULL;
+		*size = 0;
+		return 0;
+	}
+	return it->position(it, pos, size);
+}
+
+/** Keys for iterator position map. All the keys must be uint. */
+enum {
+	ITERATOR_POSITION_KEY = 0,
+};
+
+uint32_t
+iterator_position_pack_size(const char *pos, const char *pos_end,
+			    uint32_t part_count)
+{
+	assert(pos != NULL);
+	assert(part_count != 0);
+	uint32_t total = pos_end - pos;
+	total += mp_sizeof_array(part_count);
+	total += mp_sizeof_uint(ITERATOR_POSITION_KEY);
+	total += mp_sizeof_map(1);
+	total += mp_sizeof_binl(total);
+	return total;
+}
+
+void
+iterator_position_pack(const char *pos, const char *pos_end,
+		       uint32_t part_count, char *buffer)
```

Comment:
Better rename `buffer` to `packed_pos` and add `packed_pos_end` for consistency.

Reply:
Done